pkg/kv/kvserver/batcheval/cmd_delete_range.go

```diff
@@ -85,8 +86,28 @@ func DeleteRange(
 			args.Key, args.EndKey, desc.StartKey.AsRawKey(), desc.EndKey.AsRawKey())
 		maxIntents := storage.MaxIntentsPerWriteIntentError.Get(&cArgs.EvalCtx.ClusterSettings().SV)
 
-		err := storage.MVCCDeleteRangeUsingTombstone(ctx, readWriter, cArgs.Stats,
-			args.Key, args.EndKey, h.Timestamp, cArgs.Now, leftPeekBound, rightPeekBound, maxIntents)
+		if args.Predicates == nil {
+			err := storage.MVCCDeleteRangeUsingTombstone(ctx, readWriter, cArgs.Stats,
+				args.Key, args.EndKey, h.Timestamp, cArgs.Now, leftPeekBound, rightPeekBound, maxIntents)
+			return result.Result{}, err
+		}
+		maxBatchSize := h.MaxSpanRequestKeys
+		if h.MaxSpanRequestKeys == 0 {
+			maxBatchSize = math.MaxInt64
+		}
+
+		// The minimum number of keys required in a run to use a range tombstone
+		// TODO (msbutler): how to tune this?
+		defaultRangeTombstoneThreshold := int64(64)
+		resumeSpan, err := storage.PredicateMVCCDeleteRange(ctx, readWriter, cArgs.Stats,
+			args.Key, args.EndKey, h.Timestamp, cArgs.Now, leftPeekBound, rightPeekBound,
+			args.Predicates, maxBatchSize, maxRevertRangeBatchBytes, defaultRangeTombstoneThreshold)
+
+		if resumeSpan != nil {
+			reply.ResumeSpan = resumeSpan
+			reply.ResumeReason = roachpb.RESUME_KEY_LIMIT
+		}
+		// TODO(msbutler): build correct result response
```

Comment:
@erikgrinaker could you point me to example code on how to construct a proper resumer for this? Maybe it's ok to use `result.FromAcquiredLocks(h.Txn, deleted...)` like in vanilla DeleteRange?

Reply:
Nah. Range tombstones can't be transactional, so there is no `h.Txn`. Essentially, just return a `resumeKey MVCCKey` from the function if you need to bail out early, and put that in the client response. I don't think we need any further magic here?

Reply:
Resolved offline.